packages/contracts/contracts/engine/V2/GenArt721MinterBurner_PBAB.sol

```diff
@@ -215,18 +215,49 @@ contract GenArt721MinterBurner_PBAB is ReentrancyGuard {
     }
 
     /**
-     * @notice Purchases a token from project `_projectId`.
+     * @notice Purchases a token from project `_projectId` with ETH.
      * @param _projectId Project ID to mint a token on.
      * @return _tokenId Token ID of minted token
      */
     function purchase(
         uint256 _projectId
     ) public payable returns (uint256 _tokenId) {
-        return purchaseTo(msg.sender, _projectId);
+        // pass currency address as ETH
+        address _currencyAddress = address(0);
+        uint256 _maxPricePerToken = msg.value;
+
+        return
+            purchaseTo(
+                msg.sender,
+                _projectId,
+                _maxPricePerToken,
+                _currencyAddress
+            );
```

Comment:
Another way to do this that might add smol clarity and be slightly more gas efficient could be:
```
        // pass max price as msg.value, currency address as ETH
        return
            purchaseTo({
                to: msg.sender,
                _projectId: _projectId,
                _maxPricePerToken: msg.value,
                _currencyAddress: address(0)
            });
```
(should definitely stay consistent stylistically between this and the other ETH-only function though!)

Reply:
Nice! Updated 🙏